code/__HELPERS/_lists.dm

```diff
@@ -73,6 +73,8 @@
 	return FALSE
 
 //Checks for specific types in specifically structured (Assoc "type" = TRUE) lists ('typecaches')
+#define is_type_in_typecache(A, L) (length(L) && L[(ispath(A) ? A : A:type)])

```

Comment:
Behavior has been changed when A is null; it will fail the `ispath()` check and try to access `null:type`.